sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs

```diff
@@ -90,8 +91,23 @@ public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder bui
             }
 
             builder.WithTracing(b => b
+                            .AddSource("Azure.*")
                             .AddAspNetCoreInstrumentation()
-                            .AddHttpClientInstrumentation()
+                            .AddHttpClientInstrumentation(o => o.FilterHttpRequestMessage = (_) =>
+                            {
+                                // Azure SDKs create their own client span before calling the service using HttpClient
+                                // In this case, we would see two spans corresponding to the same operation
+                                // 1) created by Azure SDK 2) created by HttpClient
+                                // To prevent this duplication we are filtering the span from HttpClient
+                                // as span from Azure SDK contains all relevant information needed.
+                                var parentActivity = Activity.Current?.Parent;
+                                if (parentActivity != null && parentActivity.Source.Name.StartsWith("Azure.Core"))
```

Comment:
@lmolkova Is it ok to specify the exact name here i.e. `Azure.Core.Http.Request`?

Reply:
The source name is "Azure.Core.Http", but yes, absolutely.  Tested`parentActivity.Source.Name.Equals("Azure.Core.Http")` - ✅